### PR TITLE
fix test file generation to allow for sessions of different length

### PR DESCRIPTION
**Pull request by l-k-**
`create_test_acc_csv()` was failing whenever `Nmin` was such that one of the sleep blocks started just before the end of the recording but didn't end within the recording. Same issue for non-wear periods and activity periods.

For instance, `create_test_acc_csv(Nmin = 3000)` was attempting to create a non-wear block that ran past the end of the recording, resulting in an error:

> GGIR::create_test_acc_csv(Nmin = 3000)
Error in `[<-`(`*tmp*`, nw3, 1, value = accx) : subscript out of bounds

I modified `create_test_acc_csv.R` to test these various `Nmin` values that lead to errors.

Basically the problem was that `create_test_acc_csv()` was making sure sleep/non-wear/activity blocks started within the recording period, but it wasn't making sure that these blocks also *ended* within the recording period.

The fix ensures that no sleep block (or non-wear block, or activity block) ends after the end of the recording. If a block is too long to fully fit within a recording, it gets truncated.

### CODE REVIEW

**Comment by vincentvanhees** [comment]
thanks @l-k-, looks great and code is a even a bit tidier now.
